udev-enumerate.c

```diff
@@ -210,6 +211,54 @@ udev_enumerate_scan_devices(struct udev_enumerate *ue)
 	return ret;
 }
 
+/*
+ * Enumerate subsystems -- under /sys/modules, /sys/dev, only
+ * list the directories.
+ */
+static int
+enumerate_ssys_cb(const char *path, int type, void *arg)
+{
+	struct udev_enumerate *ue = arg;
+
+	if (type == DT_DIR) {
+		if (udev_list_insert(&ue->dev_list, path, NULL) == -1)
+			return (-1);
+	}
+	return (0);
+}
+
+LIBUDEV_EXPORT int
+udev_enumerate_scan_subsystems(struct udev_enumerate *ue)
+{
+	int ret = 0;
+
+	TRC("(%p)", ue);
+	udev_list_free(&ue->dev_list);
+
+	struct scan_ctx ctx;
+	char modules_path[DEV_PATH_MAX] = "/sys/modules/";
+	char drivers_path[DEV_PATH_MAX] = "/sys/dev/";
```

Comment:
uh so this requires `linsysfs` to be mounted at `/sys`? and I don't see the `modules` directory in `linsysfs`…

Reply:
This is just (largely) copied from systemd so that the API exists. I even forget **why** we needed this -- probably KDE Solid on FreeBSD, or possibly something NetworkManager related. The point is code that expects Linux udev will use this function. It might be fine to return an empty list, since there's no subsystems to list anyway.

My own `/sys` is a symlink to `usr/src/sys`, which probably misled me at the time. Linux `/sys` contains a `module` directory.

Reply:
hmm, are there attempts to bring NetworkManager to FreeBSD?

Reply:
We hard code paths like this. Usually kernel source is at /sys. Why do we need this functionality?